pkg/ring/kv/etcd/etcd.go

```diff
@@ -82,6 +83,9 @@ func (c *Client) CAS(ctx context.Context, key string, f func(in interface{}) (ou
 		if err != nil {
 			level.Error(util.Logger).Log("msg", "error CASing", "key", key, "err", err)
 			if !retry {
+				if resp, ok := httpgrpc.HTTPResponseFromError(err); ok && resp.GetCode() != 202 {
+					level.Error(util.Logger).Log("msg", "error CASing", "key", key, "err", err)
+				}
```

Comment:
This looks like a layering violation; why should our etcd client special case 202s?  This should go in the ha tracker - that way it would behave the same with consul too.

Reply:
How could we move this into the HA tracker? With the current structure the returning of the 202 from `f(...)`has to be an error, right?

Reply:
As discussed, lets decouple and move this to HA tracker, and not log here.